ykllvmwrap/src/memman.cc

```diff
@@ -0,0 +1,78 @@
+#include <llvm/ExecutionEngine/RTDyldMemoryManager.h>
+#include <sys/mman.h>
+
+using namespace llvm;
+
+struct AllocMem {
+  uint8_t *Ptr;
+  uintptr_t Size;
+};
+
+class MemMan : public RTDyldMemoryManager {
+public:
+  MemMan();
+  ~MemMan() override;
+
+  uint8_t *allocateCodeSection(uintptr_t Size, unsigned Alignment,
+                               unsigned SectionID,
+                               StringRef SectionName) override;
+
+  uint8_t *allocateDataSection(uintptr_t Size, unsigned Alignment,
+                               unsigned SectionID, StringRef SectionName,
+                               bool isReadOnly) override;
+
+  bool finalizeMemory(std::string *ErrMsg) override;
+  void freeMemory();
+
+private:
+  std::vector<AllocMem> code;
+  std::vector<AllocMem> data;
+};
+
+MemMan::MemMan() {}
+MemMan::~MemMan() {}
+
+uint8_t *alloc_mem(uintptr_t Size, unsigned Alignment,
+                   std::vector<AllocMem> *Vec) {
+  uintptr_t RequiredSize = Alignment * ((Size + Alignment - 1) / Alignment + 1);
+  auto Ptr = (unsigned char *)mmap(0, RequiredSize, PROT_WRITE,
```

Comment:
We should add some sort of check here maybe just `assert(Ptr != MAP_FAILED)`.

Reply:
agreed.

Reply:
Fixed in 43872de.